### PR TITLE
[ALLUXIO-2813] Remove FileSystemWorkerClient

### PR DESCRIPTION
**Pull request by calvinjia**
https://alluxio.atlassian.net/browse/ALLUXIO-2813

Some notes:
* Passing `OutStreamOptions` to `UnderFileSystemFileOutStream.create` to be consistent with other create calls (ie. `BlockOutStream.create`)
* Currently using the old FileWriteLocationPolicy's first block location to determine the file write worker, will clean this up when we decide how to handle the write/read/cache policies.

### CODE REVIEW

File: core/client/src/main/java/alluxio/client/file/FileOutStream.java [hunk 1]
```diff
@@ -99,22 +98,21 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options, FileSystemContex
...
+      try {
+        WorkerNetAddress worker = // not storing data to Alluxio, so block size is 0
+            options.getLocationPolicy().getWorkerForNextBlock(mBlockStore.getWorkerInfoList(), 0);
+        InetSocketAddress location = new InetSocketAddress(worker.getHost(), worker.getDataPort());
+        mUnderStorageOutputStream =
+            mCloser.register(UnderFileSystemFileOutStream.create(mContext, location, mOptions));
+      } catch (AlluxioException | IOException e) {
+        mCloser.close();
+        Throwables.propagateIfInstanceOf(e, IOException.class);
+        throw new IOException(e);
```
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/file/FileOutStream.java]: nit: use CommonUtils.castToIOExeception(e);

File: core/client/src/main/java/alluxio/client/file/FileOutStream.java [hunk 2]
```diff
@@ -99,22 +98,21 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options, FileSystemContex
...
+      mUnderStorageOutputStream = null;
+    } else { // Write is through to the under storage, create mUnderStorageOutputStream
+      try {
+        WorkerNetAddress worker = // not storing data to Alluxio, so block size is 0
+            options.getLocationPolicy().getWorkerForNextBlock(mBlockStore.getWorkerInfoList(), 0);
+        InetSocketAddress location = new InetSocketAddress(worker.getHost(), worker.getDataPort());
+        mUnderStorageOutputStream =
+            mCloser.register(UnderFileSystemFileOutStream.create(mContext, location, mOptions));
+      } catch (AlluxioException | IOException e) {
+        mCloser.close();
```
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/file/FileOutStream.java]: nit: use CommonUtils.closeQuietly() to avoid masking the real exception.

Comment (calvinjia) [inline core/client/src/main/java/alluxio/client/file/FileOutStream.java]: Updated, thanks.
Should this be done everywhere we call `Closer#close`, if so could you open a ticket for this?

File: core/client/src/main/java/alluxio/client/file/FileOutStream.java (hunk 1 above)
Comment (calvinjia) [inline core/client/src/main/java/alluxio/client/file/FileOutStream.java]: Updated, thanks.
Should this be done everywhere we have this kind of logic, if so could you open a ticket for this?

File: core/client/src/main/java/alluxio/client/file/FileOutStream.java (hunk 2 above)
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/file/FileOutStream.java]: I didn't create the JIRA. I am not able to find any code that does not follow this in the codebase by searching all the usecases of "closer.close()".

Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/file/FileOutStream.java]: NVM, i found one

**Comment by peisun1115** [comment]
LGTM

**Pull request merged**